src/pyuvdata/uvbeam/uvbeam.py

```diff
@@ -1710,6 +1718,9 @@ def _interp_az_za_map_coordinates(
         check_azza_domain : bool
             Whether to check the domain of az/za to ensure that they are covered by the
             intrinsic data array. Checking them can be quite computationally expensive.
+        return_basis_vector : bool
+            Whether to return the interpolated basis vector. Default is False as of
+            v3.1.1, but was previously True.
```

Comment:
This is going to break things. We need a deprecation process because this keyword didn't exist in the past. I think you should default it to None and then catch that case, issue a deprecation warning and set it to True. The deprecation warning should say that it will default to False starting in version 3.3.

Reply:
Ah yeah. So at the moment, I'm doing this deprecation stuff only in the top-level function `interp()` because these functions are private and therefore need not be API-stable. Is that OK or do you want me to do the deprecation logic everywhere?

Reply:
Oh, sorry, I didn't see that. I think that's fine.